server/start.go

```diff
@@ -167,9 +169,23 @@ is performed. Note, when enabled, gRPC will also be automatically enabled.
 				serverCtx.Logger.Info("starting ABCI without CometBFT")
 			}
 
-			return wrapCPUProfile(serverCtx, func() error {
+			err = wrapCPUProfile(serverCtx, func() error {
 				return start(serverCtx, clientCtx, appCreator, withCMT, opts)
 			})
+
+			serverCtx.Logger.Debug("received quit signal")
+
+			// wait for app.Close() and tmNode.Stop() to flush async db writes to disk
+			graceSeconds, _ := cmd.Flags().GetUint32(FlagShutdownGraceSeconds)
+			if graceSeconds > 0 {
+				serverCtx.Logger.Info(fmt.Sprintf("allow %d seconds for database to flush", graceSeconds))
```

Comment:
Nit: Don't use Sprintf to logger calls. The API accepts key/value pairs. Pass those instead :)